apps/src/applab/Exporter.js

```diff
@@ -9,6 +9,136 @@ import elementLibrary from './designElements/library';
 import exportProjectEjs from '../templates/exportProject.html.ejs';
 import exportProjectReadmeEjs from '../templates/exportProjectReadme.md.ejs';
 import logToCloud from '../logToCloud';
+import {getAppOptions} from '@cdo/apps/code-studio/initApp/loadApp';
+
+const APP_OPTIONS_WHITELIST = {
+  "levelGameName": true,
+  "skinId": true,
+  "baseUrl": true,
+  "app": true,
+  "droplet": true,
+  "pretty": true,
+  "level": {
+    "skin": true,
+    "editCode": true,
+    "embed": true,
+    "isK1": true,
+    "isProjectLevel": true,
+    "skipInstructionsPopup": true,
+    "disableParamEditing": true,
+    "disableVariableEditing": true,
+    "useModalFunctionEditor": true,
+    "useContractEditor": true,
+    "contractHighlight": true,
+    "contractCollapse": true,
+    "examplesHighlight": true,
+    "examplesCollapse": true,
+    "definitionHighlight": true,
+    "definitionCollapse": true,
+    "freePlay": true,
+    "appWidth": true,
+    "appHeight": true,
+    "sliderSpeed": true,
+    "calloutJson": true,
+    "disableExamples": true,
+    "showTurtleBeforeRun": true,
+    "autocompletePaletteApisOnly": true,
+    "textModeAtStart": true,
+    "designModeAtStart": true,
+    "hideDesignMode": true,
+    "beginnerMode": true,
+    "levelId": true,
+    "puzzle_number": true,
+    "stage_total": true,
+    "iframeEmbed": true,
+    "lastAttempt": true,
+    "submittable": true
+  },
+  "showUnusedBlocks": true,
+  "fullWidth": true,
+  "noHeader": true,
+  "noFooter": true,
+  "smallFooter": true,
+  "codeStudioLogo": true,
+  "hasI18n": true,
+  "callouts": true,
+  "channel": true,
+  "readonlyWorkspace": true,
+  "isLegacyShare": true,
+  "postMilestone": true,
+  "postFinalMilestone": true,
+  "puzzleRatingsUrl": true,
```

Comment:
At a glance, things I'd try pulling from the whitelist:

* puzzleRatingsUrl
* authoredHintViewRequestsUrl
* serverLevelId
* labUserId
* firebaseName
* firebaseAuthToken
* firebaseChannelIdSuffix
* teacherMarkdown